src/zenml/models/filter_models.py

```diff
@@ -0,0 +1,383 @@
+            return getattr(table, self.column).like(f"%{self.value}%")
+        elif self.operation == GenericFilterOps.STARTSWITH:
+            return getattr(table, self.column).startswith(f"%{self.value}%")
+        elif self.operation == GenericFilterOps.CONTAINS:
+            return getattr(table, self.column).endswith(f"%{self.value}%")
+
+
+class UUIDFilter(Filter):
+    ALLOWED_OPS: ClassVar[List[str]] = [
+        GenericFilterOps.EQUALS,
+        GenericFilterOps.STARTSWITH,
+        GenericFilterOps.CONTAINS,
+        GenericFilterOps.ENDSWITH,
+    ]
+
+    def generate_query_conditions(
+        self,
+        table: Type[SQLModel],
+    ):
+        """Generate the query conditions for the database.
+
+        Args:
+            table: The SQLModel table to use for the query creation
+
+        Returns:
+            A list of conditions that will be combined using the `and` operation
+        """
+        import sqlalchemy
+        from sqlalchemy_utils.functions import cast_if
+
+        if self.operation == GenericFilterOps.EQUALS:
+            return getattr(table, self.column) == self.value
+        elif self.operation == GenericFilterOps.CONTAINS:
+            return cast_if(getattr(table, self.column), sqlalchemy.String).like(
+                f"%{self.value}%"
+            )
+        elif self.operation == GenericFilterOps.STARTSWITH:
+            return cast_if(
+                getattr(table, self.column), sqlalchemy.String
+            ).startswith(f"%{self.value}%")
+        elif self.operation == GenericFilterOps.CONTAINS:
+            return cast_if(
+                getattr(table, self.column), sqlalchemy.String
+            ).endswith(f"%{self.value}%")
+
+
+class NumericFilter(Filter):
+    ALLOWED_OPS: ClassVar[List[str]] = [
+        GenericFilterOps.EQUALS,
+        GenericFilterOps.GT,
+        GenericFilterOps.GTE,
+        GenericFilterOps.LT,
+        GenericFilterOps.LTE,
+    ]
+
+    def generate_query_conditions(
+        self,
+        table: Type[SQLModel],
+    ):
+        """Generate the query conditions for the database.
+
+        Args:
+            table: The SQLModel table to use for the query creation
+
+        Returns:
+            A list of conditions that will be combined using the `and` operation
+        """
+        if self.operation == GenericFilterOps.EQUALS:
+            return getattr(table, self.column) == self.value
+        elif self.operation == GenericFilterOps.GTE:
+            return getattr(table, self.column) >= self.value
+        elif self.operation == GenericFilterOps.GT:
+            return getattr(table, self.column) > self.value
+        elif self.operation == GenericFilterOps.LTE:
+            return getattr(table, self.column) <= self.value
+        elif self.operation == GenericFilterOps.LT:
+            return getattr(table, self.column) < self.value
+
+
+# ---------------- #
+# PAGINATION PARAM #
+# -----------------#
+
+
+@dataclass
```

Comment:
Any specific reason for using `dataclass` here instead of making this a pydantic model as all the others?

Reply:
nope, copy pasted from somewhere

Reply:
its gone now